cep-20.2.md

```diff
@@ -0,0 +1,831 @@
+    # force the file type of the given files to be TEXT or BINARY
+    # for prefix replacement
+    force_file_type:
+      # force TEXT file type
+      # was `???`
+      text: [glob]
+      # force binary file type
+      # was `???`
+      binary: [glob]
+
+    # ignore all or specific files for prefix replacement
+    # was `ignore_prefix_files`
+    ignore: bool | [path] (defaults to false)
+
+    # wether to detect binary files with prefix or not
+    # was `detect_binary_files_with_prefix`
+    ignore_binary_files: bool (defaults to true on Unix and (always) false on Windows)
+
+  # settings for shared libraries and executables
+  dynamic_linking:
+    # linux only, list of rpaths (was rpath)
+    rpaths: [path] (defaults to ['lib/'])
+
+    # wether to relocate binaries or not. If this is a list of paths, then
+    # only the listed paths are relocated
+    binary_relocation: bool (defaults to true) | [glob]
+
+    # Allow linking against libraries that are not in the run requirements
+    # (was `missing_dso_whitelist`)
+    missing_dso_allowlist: [glob]
+
+    # Allow runpath / rpath to point to these locations outside of the environment
+    # (was `runpath_whitelist`)
+    rpath_allowlist: [glob]
+
+    # what to do when detecting overdepending
+    overdepending_behavior: OneOf<"ignore" | "error"> # (defaults to "error")
+
+    # what to do when detecting overlinking
+    overlinking_behavior: OneOf<"ignore" | "error"> # (defaults to "error")
+
+  # REMOVED:
+  # pre-link: string (was deprecated for a long time)
+  # Whether to include the recipe or not in the final package - should be specified on command line or other config file?
+  # noarch_python: bool
+  # features: list
+  # msvc_compiler: str
+  # requires_features: dict
+  # provides_features: dict
+  # preferred_env: str
+  # preferred_env_executable_paths: list
+  # disable_pip: bool
+  # marked as "still experimental"
+  # pin_depends: Enum<"record" | "strict">
+  # overlinking_ignore_patterns: [glob]
+  # defaults to patchelf (only cudatoolkit is using `lief` for some reason)
+  # rpaths_patcher: None
+  # post-link: path
+  # pre-unlink: path
+  # pre-link: path
+```
+
+### Script section
+
+```yaml
+script:
+  # the interpreter to use for the script
+  interpreter: string  # defaults to bash on UNIX and cmd.exe on Windows
+  # the script environment. You can use Jinja to pass through environment variables
+  # with the `env` key (`${{ env.get("MYVAR") }}`).
+  env: {string: string}
+  # secrets that are set as env variables but never shown in the logs or the environment
+  # The variables are taken from the parent environment by name (e.g. `MY_SECRET`)
+  secrets: [string]
+  # The file to use as the script. Automatically adds `bat` or `sh` to the filename
+  # on Windows or UNIX respectively (if no file extension is given).
+  file: string  # build.sh or build.bat
+  # A string or list of strings that is the script contents (mutually exclusive with `file`)
```

Comment:
We should define what should happen if both `file` and `content` appear in the same recipe specification file. (Error out vs. one taking priority)